src/context.c

```diff
@@ -91,20 +93,21 @@ ly_ctx_new(const char *search_dir)
     ctx->models.used = 0;
     ctx->models.size = 16;
     if (search_dir) {
-        cwd = get_current_dir_name();
-        if (chdir(search_dir)) {
-            LOGERR(LY_ESYS, "Unable to use search directory \"%s\" (%s)",
-                   search_dir, strerror(errno));
-            goto error;
-        }
-        ctx->models.search_paths = malloc(2 * sizeof *ctx->models.search_paths);
+        ctx->models.search_paths = malloc(sizeof *ctx->models.search_paths);
```

Comment:
This allocation shouldn't be needed, just keep search_paths NULL and ly_ctx_set_searchdir() will do the rest.